garden_ai/local_data.py

```diff
@@ -45,138 +48,151 @@ def _get_user_email() -> str:
     return str(maybe_email) if maybe_email else "unknown"
 
 
-# TODO: eliminate redundant ways to write a garden once we fix the deserialization issue.
-def put_local_garden_metadata(garden_metadata: Dict) -> None:
-    """Helper: write a record to 'local database' for a given Garden
+def _put_resource_from_metadata(
+    resource_metadata: Dict, resource_type: ResourceType
+) -> None:
+    data = _read_local_db()
+    resources = data.get(resource_type.value, {})
+    resources[str(resource_metadata["uuid"])] = resource_metadata
+    data[resource_type.value] = resources
+    _write_local_db(data)
 
-    Overwrites any existing entry with the same uuid in ~/.garden/data.json.
 
-    Parameters
-    ----------
-    garden_metadata : Dict
-        Garden metadata in the format given by Garden.json()
+def _put_resource_from_pydantic(resource: Union[Garden, Pipeline]) -> None:
```

Comment:
🔮🎭 Ahoy there, matey! 🏴‍☠️ How'd ya feel about renaming this here function to _put_resource_from_obj ⚓️ instead of _put_resource_from_pydantic? 🤔 Arrr, just 'cause we might decide that the Pipeline interface we be wantin' can't be a Pydantic model or somethin', ya know? 🏝☠️💡

Reply:
Sure!